Server/Scripts/configure.ps1

```diff
@@ -3,167 +3,126 @@ $env:OCTOPUS_INSTANCENAME=$OctopusInstanceName
-      mkdir C:\Octopus\Logs | Out-Null
-    }
-    mv C:\Octopus\LogsTemp\* C:\Octopus\Logs
-    Remove-Item C:\Octopus\LogsTemp -Recurse -Force
-    Write-Log "moved"
+function Configure-Database() {
+  if($null -eq $sqlDbConnectionString){
+    Write-Error "Environment variable sqlDbConnectionString required"
+    Exit 2
   }
+  $maskedConnectionString = "$sqlDbConnectionString;" -replace "password=.*?;", "password=###########;"
+  Write-Log " - using database '$maskedConnectionString'"
+  Write-Log "Creating Octopus Deploy database ..."
+  $args = @('database','--instance', $OctopusInstanceName, '--connectionString', $sqlDbConnectionString, '--create')
+  if ($null -ne $masterKey) {
+    $args += @('--masterkey', $masterKey)
+  }
+  Execute-Command $Exe $args @($masterKey, $sqlDbConnectionString)
+  Export-MasterKey
+}
 
-  function Configure-OctopusDeploy() {
-    $port=81
-    $listenPort=10943
-    $webListenPrefixes = "http://localhost:$port"
-
-    Write-Log "Configure Octopus Deploy"
+function Configure-Paths() {
+  Write-Log "Configuring Paths ..."
+  Execute-Command $Exe @('path', '--instance', $OctopusInstanceName,
+    '--nugetRepository', 'C:\Repository',
+    '--artifacts', 'C:\Artifacts',
+    '--taskLogs', 'C:\TaskLogs'
+  )
+}
 
-    Write-Log "Creating Octopus Deploy database ..."
-    $args = @(
-      'database',
-      '--console',
+function Configure-Node() {
+  Write-Log "Configuring Octopus Deploy instance with default options ..."
+  $args = @(
+      'configure',
       '--instance', $OctopusInstanceName,
-      '--connectionString', $sqlDbConnectionString,
-      '--create'
+      '--upgradeCheck', 'True',
+      '--webForceSSL', 'False',
+      '--webListenPrefixes', "http://localhost:81",
+      '--commsListenPort', 10943
     )
-    if ($masterKeySupplied) {
-      $args += '--masterkey'
-      $args += $masterKey
+
+  if($env:EnableUsage -eq "Y" -or $env:EnableUsage -eq "y" ) {
+    $args += @('--upgradeCheckWithStatistics', 'True')
+  } else {
+    $args += @('--upgradeCheckWithStatistics', 'False')
+  }
+
+  if($null -ne $env:ServerNodeName) {
+    $args += @('--serverNodeName', $env:ServerNodeName)
+  }
+  
+  Execute-Command $Exe $args
+
+  if($env:EnableMetrics -eq "true") {
+    Write-Log "Enabling metrics logging"
+    Execute-Command $Exe @('metrics', '--instance', $OctopusInstanceName, '--enable', '--tasks', 'true', '--webapi', 'true')
+  }
+}
+
+function Configure-Admin() {
+
+    if(($null -eq $masterKey) -and ($null -eq $octopusAdminUsername)) {
+      Write-Log "WARNING: No admin username provided, using default."
+      $octopusAdminUsername="admin";
     }
-    Execute-Command $Exe $args @($masterKey, $sqlDbConnectionString)
-
-    Write-Log "Configuring Octopus Deploy instance with default options ..."
-    Execute-Command $Exe @(
-        'configure',
-        '--console',
-        '--instance', $OctopusInstanceName,
-        '--home', 'C:\Octopus',
-        '--serverNodeName', $ServerNodeName,
```

Comment:
Glad to see this gone... It's a dangerous beastie.
Oh. It's not gone, its just configurable.